### PR TITLE
[MM-29408] -migrate-file-upload-refs

### PR DESCRIPTION
**Pull request by sridhar02**
#### Summary
Migrate refs to new syntax of file_upload_settings.jsx
#### Ticket Link
  Fixes https://github.com/mattermost/mattermost-server/issues/15847

### CODE REVIEW

File: components/admin_console/file_upload_setting.jsx [hunk 1]
```diff
@@ -32,10 +32,11 @@ export default class FileUploadSetting extends Setting {
             fileName: null,
             serverError: props.error,
         };
+        this.fileInputRef = React.createRef();
     }
 
     handleChange = () => {
-        const files = this.refs.fileInput.files;
+        const files = this.fileInput.current.files;
```
Comment (Willyfrog) [inline components/admin_console/file_upload_setting.jsx:39]: shouldn't this be `this.fileInputRef`? I see it also in some other parts of the file

**Review by Willyfrog — CHANGES_REQUESTED** [review]
Hi there @sridhar02 , thanks for contributing.
there is a second reference in this file: `upload_button`, could you also migrate that ref?

File: components/admin_console/file_upload_setting.jsx (hunk 1 above)
Comment (sridhar02) [reply components/admin_console/file_upload_setting.jsx]: Resolved it.

**Review by nevyangelova — APPROVED** [review]
Thanks! LGTM!

**Review by Willyfrog — CHANGES_REQUESTED** [review]
looking good, I just noticed I forgot to mention to also remove the eslint rule for allowing string-refs, could you remove it? is this one

```/* eslint-disable react/no-string-refs */```

**Review by Willyfrog — APPROVED** [review]
thanks!

**Review by jgilliam17 — APPROVED** [review]
Thank you @sridhar02 
Tested, looks good to merge.
- Verified/tested around file upload - as expected.

**Pull request merged**